src/ch/blinkenlights/android/vanilla/SongTimeline.java

```diff
@@ -580,10 +582,18 @@ public Song getSong(int delta)
 				return null;
 			} else if (pos == size) {
 				if (mFinishAction == FINISH_RANDOM) {
-					song = MediaUtils.getRandomSong(mContext, mShuffleMode == SHUFFLE_ALBUMS);
-					if (song == null)
+
+					final List<Song> songs = MediaUtils.getRandomSongs(mContext, mShuffleMode == SHUFFLE_ALBUMS);
+					if (songs.size() == 0) {
 						return null;
-					timeline.add(song);
+					}
+
+					for (Song newSong : songs) {
+						timeline.add(newSong);
+					}
+
+					song = songs.get(songs.size() - 1);
```

Comment:
Shouldn't this be `songs.get(0)` ?
The current code returns the last song we added while the caller just asked for the next song (which would be the first in our list)

Reply:
Changed back to getting 0.